### ISSUE TITLE
lintDirtyModulesOnly doesnt seem to be working

### ISSUE THREAD

**Issue by thasmo** [webpack-contrib/eslint-webpack-plugin#2]
- Operating System: Windows 10 Pro 64 1909
- Node Version: v10.16.3
- NPM Version: 6.12.1
- webpack Version: 5.0.0-beta.4
- eslint-webpack-plugin Version: 0.0.1

### Expected Behavior
`lintDirtyModulesOnly: true` prints linting messages for dirty/changed files only when in watch mode.

### Actual Behavior
Using `lintDirtyModulesOnly: false` prints linting messages for all files as expected but `lintDirtyModulesOnly: true` doesn't print anything at all.

### Code
```ts
const lint = new EslintWebpackPlugin({
    extensions: ['.ts', '.tsx'],
    lintDirtyModulesOnly: true,
});
```

Not sure what else I can provide, project is not public so I can't post a link or something. Just let me know if I can have a look at anything specific. Thanks!

**Comment by thasmo** [comment]
Could it be that there's a mismatch in how the `files` option is used in `EslintWebpackPlugin` and `LintDirtyModulesPlugin`? Afaik `EslintWebpackPlugin` can not handle glob values but `LintDirtyModulesPlugin` expects it to be a glob value.

**Comment by ricardogobbosouza** [comment]
Hi @thasmo
You need to specify where your files are, i'll try to improve it

try this:
```js
const lint = new EslintWebpackPlugin({
    files: 'src/**/*.ts',
    extensions: ['.ts', '.tsx'],
    lintDirtyModulesOnly: true,
});
````

**Comment by thasmo** [comment]
@ricardogobbosouza, thanks! When using proper values for `files` and `extensions` I get this error:
```
Error: No files matching 'C:\Users\user\Projects\project\assets\base.ts' were found.
```
There is no file `C:\Users\user\Projects\project\assets\base.ts` but there's a directory `C:\Users\user\Projects\project\assets\base`.

The full trace is this:
```
｢wdm｣: Error: No files matching 'C:\Users\u\Projects\p\assets\base.ts' were found.
at FileEnumerator.iterateFiles (C:\Users\u\Projects\p\node_modules\eslint\lib\cli-engine\file-enumerator.js:276:23)
at iterateFiles.next (<anonymous>)
at CLIEngine.executeOnFiles (C:\Users\u\Projects\p\node_modules\eslint\lib\cli-engine\cli-engine.js:747:48)
at linter (C:\Users\u\Projects\p\node_modules\eslint-webpack-plugin\dist\linter.js:23:24)
at LintDirtyModulesPlugin.apply (C:\Users\u\Projects\p\node_modules\eslint-webpack-plugin\dist\LintDirtyModulesPlugin.js:39:27)
at compiler.hooks.watchRun.tapAsync (C:\Users\u\Projects\p\node_modules\eslint-webpack-plugin\dist\index.js:39:19)
at Hook.eval [as callAsync] (eval at create (C:\Users\u\Projects\p\node_modules\webpack\node_modules\tapable\lib\HookCodeFactory.js:33:10), <anonymous>:20:1)
at compiler.cache.endIdle.err (C:\Users\u\Projects\p\node_modules\webpack\lib\Watching.js:62:33)
at C:\Users\u\Projects\p\node_modules\webpack\lib\HookWebpackError.js:69:3
at Hook.eval [as callAsync] (eval at create (C:\Users\u\Projects\p\node_modules\webpack\node_modules\tapable\lib\HookCodeFactory.js:33:10), <anonymous>:15:1)
at Cache.endIdle (C:\Users\u\Projects\p\node_modules\webpack\lib\Cache.js:140:22)
at Watching._go (C:\Users\u\Projects\p\node_modules\webpack\lib\Watching.js:60:23)
at Watching._invalidate (C:\Users\u\Projects\p\node_modules\webpack\lib\Watching.js:216:9)
at watcher.compiler.watchFileSystem.watch (C:\Users\u\Projects\p\node_modules\webpack\lib\Watching.js:182:11)
at Watchpack.watcher.once (C:\Users\u\Projects\p\node_modules\webpack\lib\node\NodeWatchFileSystem.js:59:4)
at Object.onceWrapper (events.js:286:20)
```

**Comment by swrobel** [comment]
@ricardogobbosouza so `context` isn't working? I'm using only that option w/ the stylelint plugin & it works great. Also, I don't see `extensions` mentioned as an option in the README

**Comment by thasmo** [comment]
@swrobel, `extensions` is an option of the [CLIEngine](https://eslint.org/docs/developer-guide/nodejs-api#cliengine):
```
extensions - An array of filename extensions that should be checked for code. The default is an array containing just ".js". Corresponds to --ext. It is only used in conjunction with directories, not with filenames, glob patterns or when using executeOnText().
```
Maybe that also means it's not used when using a glob on `files`.

**Comment by swrobel** [comment]
OK, great. Yes, the combination of `context` & `extensions` is working for me, although `lintDirtyModulesOnly` seems broken.

**Comment by ricardogobbosouza** [comment]
@thasmo, @swrobel thanks for reporting...I will analyze.

@swrobel as @thasmo  said, extensions is a CLIEngine option.

**Comment by ricardogobbosouza** [comment]
Hi @thasmo, @swrobel sorry for the delay.

Fixed `lintDirtModulesOnly` on [v0.1.0](https://github.com/webpack-contrib/eslint-webpack-plugin/releases/tag/v0.1.0)

Try this:
```js
const lint = new EslintWebpackPlugin({
    extensions: ['.ts', '.tsx'],
    lintDirtyModulesOnly: true,
});
```

**Comment by swrobel** [comment]
Thanks @ricardogobbosouza I forgot to come back here & reply that I tested the new release & it works great.

**Comment by immortal-loop** [comment]
Hi @ricardogobbosouza 

I am facing the same issue with 
`		"eslint-webpack-plugin": "^1.0.0",`

if passing false to `lintDirtyModulesOnly` is printing all the errors, passing true does not show any error

this is the config that I am using
```

        new ESLintPlugin({
          files: './src/**/*.(js|jsx)',
          extensions: ['.js', '.jsx'],
          lintDirtyModulesOnly: true,
        }),
```

please help
